controllers/appgroup_controller.go

```diff
@@ -526,35 +540,55 @@ func (r *ApplicationGroupReconciler) cleanupWorkflow(ctx context.Context, logr l
 		for _, node := range wf.Status.Nodes {
 			nodes[node.ID] = node
 		}
-		graph, err := workflow.Build(g.Name, nodes)
-		if err != nil {
-			logr.Error(err, "failed to build the wf status DAG")
+		rwf := &v1alpha12.Workflow{
+			ObjectMeta: v1.ObjectMeta{Labels: make(map[string]string)},
```

Comment:
Are you using the labels field anywhere?

Reply:
It was leftover from copy/paste. Thanks.